File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const mapped = new Map()
    for(let i = 0; i < nums.length; i++) {
        const comp = target - nums[i]
        if(mapped.has(comp)) {
            return [i, mapped.get(comp)]
        } else {
            mapped.set(nums[i], i)
        }
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        int[] result = new int[2];
        Map<Integer, Integer> mapped = new HashMap<>();
        for(int i = 0; i < nums.length; i++) {
            int comp = target - nums[i];
            if(mapped.containsKey(comp)) {
                result[1] = i;
                result[0] = mapped.get(comp);
                return result;
            }
            mapped.put(nums[i], i);
        }
        return result;
    }
}